### DISCUSSION TITLE
Issue with datm_cdeps_mx025_cfsr

### DISCUSSION THREAD

**Discussion by mjhossen** [ufs-community/ufs-weather-model#2403] [Q&A]
I am using a ufs weather model (datm-mom6-cice6) in my DA system built with GDASApp and soca-science workflow. I plot MOM.res.nc produced by regression test - datm_cdeps_mx025_cfsr that uses data in 20111001. The plot (attached) looks suspicious to me.  I am trying to figure out where the problem is. 
Can you please make your comments. Do you think the model output is right? 
<img width="676" alt="Screenshot 2024-08-20 at 3 05 08 PM" src="https://github.com/user-attachments/assets/f7c203ae-46e2-4e62-9eda-5dc39041aabe">

**Comment by jiandewang** [comment]
@mjhossen how long did you run the model ? You may also plot the cross-section at equator (one from your starting IC, one from your results). Also only plot from surface to 300m will make it easy for people to look at them.

**Comment by DeniseWorthen** [comment]
@mjhossen Are you wondering about, for example, the sharp black "spike" which show up at around 40N? Are you sure that isn't just reflecting the bottom bathy? MOM restarts fill all the values deeper than the last valid depth with the value of the last valid depth.

**Reply by jiandewang** [reply]
"MOM restarts fill all the values deeper than the last valid depth with the value of the last valid depth." 
I think they are filled with zero

**Reply by DeniseWorthen** [reply]
See the following from the RT MOM restart at 0E.

<img width="400" alt="Screenshot 2024-08-21 at 10 33 16 AM" src="https://github.com/user-attachments/assets/dc726d63-39aa-4aa0-8eb5-99a7bb263243">

**Comment by mjhossen** [comment]
I have started my experiment with initial restart file at 2019120100 (plotted).<img width="672" alt="Screenshot 2024-08-20 at 10 59 32 AM" src="https://github.com/user-attachments/assets/fa89f378-ce5a-4a7e-8e04-d75bf0fb4291">
After first cycle the vertical temp profile looks very different:
<img width="674" alt="Screenshot 2024-08-20 at 10 58 57 AM" src="https://github.com/user-attachments/assets/fc444628-8049-4466-8a09-c8d14494b31f">
@DeniseWorthen I am worried about mixed depth (about 200m). 
I also compare the model output with argo data. They look very different.
<img width="383" alt="Screenshot 2024-08-08 at 4 16 41 PM" src="https://github.com/user-attachments/assets/d87d4f42-b485-40d8-9f11-c52da6df8810">

**Comment by jiandewang** [comment]
by comparing the two plots at 30S, apparently something is wrong (most likely in DA) after 6hr run. Will you be able to turn off DA in your run ?

**Reply by mjhossen** [reply]
Yes, I can. Do you mean to run only forcast for another 6 hrs?

**Reply by mjhossen** [reply]
Oops! I should mention that I do not do any DA yet. This is Free run without assimilation.

**Reply by jiandewang** [reply]
yes, in this way we want to make sure there is no big difference between 6hr run and its IC

**Comment by jiandewang** [comment]
let me repeat this run using UWM rt.sh, will get back to you later

**Reply by mjhossen** [reply]
Thank you!

**Comment by jiandewang** [comment]
@mjhossen  I ran this job using latest UWM, but don't see big difference between IC and 12hr restart. You may check them on HERA
/scratch1/NCEPDEV/stmp2/Jiande.Wang/FV3_RT/rt_3432713/datm_cdeps_mx025_cfsr_intel

**Reply by mjhossen** [reply]
Thanks. I will have a look.

**Comment by mjhossen** [comment]
@jiandewang I plot  I.C and 12hr restart from /scratch1/NCEPDEV/stmp2/Jiande.Wang/FV3_RT/rt_3432713/datm_cdeps_mx025_cfsr_intel. Looks different! It could be a bug in my plotting script. What do you think? Did you plot them? Can you please share your plotting script?
<img width="671" alt="Screenshot 2024-08-21 at 12 19 42 PM" src="https://github.com/user-attachments/assets/26573070-c6bb-4701-8ab7-06085797a2b0">
<img width="668" alt="Screenshot 2024-08-21 at 12 20 25 PM" src="https://github.com/user-attachments/assets/2907991a-fae1-4468-9806-d2a7595164f8">

**Reply by jiandewang** [reply]
So the first plot is from IC while the 2nd one is from 12hr restart file, both of them are from my run, right, let me use my own way to do the plotting and get back to you.

**Reply by mjhossen** [reply]
Yes.

**Comment by jiandewang** [comment]
this is from 6-12hr averaged ocean output
![image](https://github.com/user-attachments/assets/d26af687-775e-4ae1-958b-c0ac07390ef8)
it is close to your IC plot. I don't have a reliable way to plot 12hr restart file at this moment.

**Reply by mjhossen** [reply]
Looks great! So, just to make sure the plot is RESTART/20111001.120000.MOM.res.nc. Am I right?

**Comment by jiandewang** [comment]
no, it is /scratch1/NCEPDEV/stmp2/Jiande.Wang/FV3_RT/rt_3432713/datm_cdeps_mx025_cfsr_intel/ocn_2011_10_01_09.nc
this is 6-12hr averaged results, 20111001.120000.MOM.res.nc is the final snapshot at 12hr but I don't have a reliable way to plot that.

**Comment by mjhossen** [comment]
@jiandewang I also got the similar plot for ocn_2011_10_01_09.nc. It confirms that my plotting script is working correctly. So the problem could be MOM.res.nc. Is there any way to fix this issue-MOM6 restart file?

<img width="672" alt="Screenshot 2024-08-21 at 3 19 14 PM" src="https://github.com/user-attachments/assets/91a423cd-9528-4377-8de6-531f538ed141">

**Reply by jiandewang** [reply]
we need to confirm whether thers is problem in restart file or not, this may take some time as the file fomat/metadata make it hard to do a clean plot (unlike the ocn*nc file that is straightforward for us to plot).

**Reply by mjhossen** [reply]
Sounds good! Thank you.

**Comment by jiandewang** [comment]
@mjhossen sorry I forgot to mention that this is density layer model so in the restart file
Temp(Time, Layer, lath, lonh)
Layer = 1, 3, 5, 7, 9, 11, 13, 15.005, 17.015, 19.03, 21.055, 23.095, 25.16,
    27.255, 29.385, 31.565, 33.81, 36.135, 38.56, 41.105, 43.795, 46.655,
    49.715, 53.015, 56.6, 60.515, 64.805, 69.525, 74.74, 80.515, 86.92,
    94.04, 101.96, 110.77, 120.575, 131.485, 143.615, 157.095, 172.06,
    188.655, 207.035, 227.365, 249.82, 274.585, 301.86, 331.855, 364.795,
    400.915, 440.46, 483.69, 530.885, 582.335, 638.345, 699.24, 765.355,
    837.04, 914.66, 998.605, 1089.28, 1187.105, 1292.525, 1406, 1528.01,
    1659.05, 1799.64, 1950.325, 2111.665, 2284.245, 2468.675, 2665.585,
    2875.63, 3099.485, 3337.85, 3591.46, 5111.085
here the value of Layer doesn't really mean when you set z=1 then all grid points in the 1st layer is at 1m. 
the z dimension in variable is misleading (as this is not a z level model). The reason your plot looks reasonable in IC restart is because this file is from ocean DA (years ago's stuff) that used z level in its vertical configuration. 
when you use ocn*nc file the plot looks good this is because we interpolate the 75 density layer value to 40 z level.

**Reply by mjhossen** [reply]
FYI, I have some results from old ufs model (MOM.res.nc). The MOM6 restart file looks good
<img width="675" alt="Screenshot 2024-08-21 at 5 20 58 PM" src="https://github.com/user-attachments/assets/e69e6f8f-3883-4e8a-8d66-e602e9ac638c">
However, MOM6 restart file with new ufs model is the problem. I believe there is something wrong with MOM6 restart file.

**Reply by mjhossen** [reply]
Also, please see the layout from old and new ufs restart netcdf file. 
<img width="570" alt="Screenshot 2024-08-21 at 5 29 53 PM" src="https://github.com/user-attachments/assets/a3f010df-f857-433f-8d4b-8b058d732de0">

<img width="545" alt="Screenshot 2024-08-21 at 5 24 08 PM" src="https://github.com/user-attachments/assets/3216f9c4-4843-4bc2-9032-f09ab5b13415">

**Reply by jiandewang** [reply]
> FYI, I have some results from old ufs model (MOM.res.nc). The MOM6 restart file looks good <img alt="Screenshot 2024-08-21 at 5 20 58 PM" width="675" src="https://private-user-images.githubusercontent.com/65732322/360123722-e69e6f8f-3883-4e8a-8d66-e602e9ac638c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQyNzYyNTUsIm5iZiI6MTcyNDI3NTk1NSwicGF0aCI6Ii82NTczMjMyMi8zNjAxMjM3MjItZTY5ZTZmOGYtMzg4My00ZThhLThkNjYtZTYwMmU5YWM2MzhjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIxVDIxMzIzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1YmJhMmNmMDc3Y2I0ZmZkZWZjZGI0MmYyYWVlMGZhNWE1OTBiYWQxZjgwNGZiMDUzZWFhNTA5MTEyNmYzZGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OMztFMDJhUlkMsMhILYfNocngL3Ol9FubW9PdZRU4zw"> However, MOM6 restart file with new ufs model is the problem. I believe there is something wrong with MOM6 restart file.

this one must come from z level configured run

**Comment by jiandewang** [comment]
so in short, there is no reliable way to plot vertical cross section temperature in the 12hr restart file as each grid is at different depth

**Comment by DeniseWorthen** [comment]
@mjhossen Where is your run directory?

**Reply by mjhossen** [reply]
Here it is: /work/noaa/aoml-osse/jhossen/OSEwGdasapp/FR_hofx

**Reply by DeniseWorthen** [reply]
I can't see into your aoml-osse directory. Do you have any publicly accessible stmp space?

**Comment by jiandewang** [comment]
@mjhossen 
I have two sample IC restart files here

(2) /scratch1/NCEPDEV/climate/role.ufscpara/IC/HR3marine/2020073100/ocean/MOM.res.nc
(1) /scratch1/NCEPDEV/climate/role.ufscpara/IC//CPC3Dvar/2016070100/ocn/025/MOM.res.nc

![image](https://github.com/user-attachments/assets/f686ceb2-c9e9-4fff-9ec6-b6ac2c99016a)

here left is from (1), right is from (2)

you can see left panel looks reasonable but right panle looks bad
but (1) is from CPC ocean DA which used z-level in model configuration while (2) is from replay IC which used density level, although both of them z=75.

for your run and my run, IC is from (1). Once model started to run, the saved restart files will be in the format of (2) as we use density level in UFS. There is no simple way to plot cross section in z for restart files in the style of (2).

**Reply by mjhossen** [reply]
So, you are saying that MOM restart output is good; no issue with them at all. 
If this is true, I agree with that as my obs_out plot provides very good matching with the observations. 
<img width="392" alt="Screenshot 2024-08-16 at 11 29 53 AM" src="https://github.com/user-attachments/assets/49abfd45-9d52-4059-9a93-03383a253c0b">
<img width="392" alt="Screenshot 2024-08-16 at 11 32 07 AM" src="https://github.com/user-attachments/assets/b89a8077-0ceb-4073-9f4a-4e96d5e41e9c">

**Reply by jiandewang** [reply]
Yes. In the future when you compare model results with OBS always use ocn*nc file.

**Reply by mjhossen** [reply]
Thank you so much for your time and explanation.

**Comment by gspetro-NOAA** [comment]
Closing this issue since it seems to be resolved. Please feel free to post a new issue if follow-on questions come up, or request to reopen this issue.
